homeassistant/components/home_connect/__init__.py

```diff
@@ -259,20 +273,9 @@ async def update_all_devices(hass: HomeAssistant, entry: ConfigEntry) -> None:
     data = hass.data[DOMAIN]
     hc_api = data[entry.entry_id]
 
-    device_registry = dr.async_get(hass)
     try:
         await hass.async_add_executor_job(hc_api.get_devices)
         for device in hc_api.devices:
-            device_entry = device_registry.async_get_or_create(
-                config_entry_id=entry.entry_id,
-                identifiers={(DOMAIN, device.appliance.haId)},
-                name=device.appliance.name,
-                manufacturer=device.appliance.brand,
-                model=device.appliance.vib,
-            )
```

Comment:
It seems that the error is caused because `test_dianostics` doesn't load any platform and because I deleted the above lines, so devices are not created when no platform is loaded. 
If a platform is loaded, it creates the device while creating the device:
https://github.com/home-assistant/core/blob/137b9d2780be8d7e6d4d9ceadecbb54ff243e249/homeassistant/components/home_connect/entity.py#L27-L32

Should I force a platform for diagnostics testing?